compliance/index/v4/node.go

```diff
@@ -157,8 +160,29 @@ func filterPackages(_ context.Context, pck []*claircore.Package) []*claircore.Pa
 	return filtered
 }
 
+func createClient() (*http.Client, error) {
+	clientCert, err := mtls.LeafCertificateFromFile()
+	if err != nil {
+		return nil, errors.Wrap(err, "obtaining client certificate")
+	}
+	tlsConf := &tls.Config{
+		InsecureSkipVerify: true,
+		Certificates:       []tls.Certificate{clientCert},
+	}
+	return &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsConf,
+			Proxy:           proxy.FromConfig(),
+		},
+		Timeout: 30 * time.Second,
+	}, nil
```

Comment:
I see we have a default timeout here - which is fine - but are we also making this call with a context, so that it is cancellable in case something goes wrong? (you remember when Sensor calling docker registries was hanging forever cause we were not canceling a context?)

Reply:
This client is directly handed off to ClairCore, so we don't have direct control over what is done there.
I traced it and found that they are not checking the context for cancellation.
ClairCore ends up calling golangs `NewRequestWithContext` and the clients `Do` which also don't check for cancellations IIUC.
This is certainly something we might want to discuss with the CC folks.

Reply:
This might have influence on our tests in CI - need to keep in mind for the future that the calling client may need to wait for 30s (or in some cases forever) to unblock compliance container.

Reply:
The alternative would be to replace the [calling code of the repository scanner](https://github.com/stackrox/stackrox/pull/13015/files#diff-6fc1ea8e64a161479830ac48435b16e14ff2aca9b3e2884b04ce3315b184945dR202-R208) to handle the context, right? 
```go
	type result struct {
		repos []*claircore.Repository
		err   error
	}
	res := make(chan result)
	defer close(res)

	go func() {
		re, err := sc.Scan(ctx, l)
		res <- result{
			repos: re,
			err:   err,
		}
	}()

	var reps []*claircore.Repository

	select {
	case <- ctx.Done():
		log.Warnf("Repository Scanner context canceled")
	case r := <-res:
		if r.err != nil {
			return nil, errors.Wrap(err, "failed to scan repositories")
		}
		if r.repos != nil {
			log.Infof("Num repositories found: %v", len(r.repos))
			reps = r.repos
		}
	}
```

Reply:
After discussing on Slack, I see that the client will be used to make a Get call that has a context that is cancelable and has timeout (it is in the ClairCore code). So all the dangers from my first message in this thread do not apply here.